buildutils/src/clean-packages.ts

```diff
@@ -13,7 +13,7 @@ exitOnUncaughtException();
 // Get all of the packages.
 const basePath = path.resolve('.');
 const baseConfig = readJSONFile(path.join(basePath, 'package.json'));
-const packageConfig = baseConfig.workspaces.packages;
+const packageConfig = baseConfig.workspaces;
```

Comment:
Can you clarify this change? I wonder if the failing CI has something to do with it.

Reply:
The attribute "packages" doesn't exist anymore, and when I tried to run `jlpm clean` there was an error. See https://github.com/jupyterlab/jupyterlab/blob/c570641c70899b83e5cbf93e8fad8bb275c4be85/package.json#L6-L7

I don't think the CI failures are related to this change.

Reply:
Hi @krassowski,

The integrity failures come from the `@jupyterlab/nbconvert-css` package. Some how, when building Jupyterlab from this PR, the `@jupyterlab/nbconvert-css` package generates
* jupyterlab/packages/nbconvert-css/style/e3b0c44298fc1c149afb.eot
* jupyterlab/packages/nbconvert-css/style/e3b0c44298fc1c149afb.ttf
* jupyterlab/packages/nbconvert-css/style/e3b0c44298fc1c149afb.woff
* jupyterlab/packages/nbconvert-css/File style/e3b0c44298fc1c149afb.woff2

The `@jupyterlab/nbconvert-css` package shouldn't generate those files, see the `webpack.conf.js` https://github.com/jupyterlab/jupyterlab/blob/694d3801fefc23d223fd404390d51f6403a5775b/packages/nbconvert-css/webpack.config.js#L31-L51

I reverted the change made to the script clean packages, but it is failing yet.

Do you know what could be the cause?

Reply:
`null-loader` is deprecated in Webpack 5, don't know why it works up to now.

Reply:
I've tired a few things but could not narrow it down. Really puzzling :question: .

Reply:
Ok I figured out what is happening but does not yet have a fix.

So within this PR, `@jupyterlab/application` was added as dependency of `@jupyterlab/lsp`. This implies that the associated styles are added too and in particular the font awesome styles 

https://github.com/jupyterlab/jupyterlab/blob/d58fcbc3a45677edf427fdd2878e597918655314/packages/application/style/index.js#L7-L8

By dependency propagation (through `@jupyterlab/notebook`), `@jupyterlab/application` are now included when building nbconvert-css. But we were explicitly not importing all application styling:

https://github.com/jupyterlab/jupyterlab/blob/d58fcbc3a45677edf427fdd2878e597918655314/packages/nbconvert-css/raw.js#L6

So due to the new dependency cycle, `@jupyterlab/nbconvert-css` is including fonts that were not included previously. Hence we were never using the rules with `null-loader`.

---

Now I'll look for a fix

Reply:
In fact (and partly related to the discussion below about https://github.com/jupyterlab/jupyterlab/issues/14782), I guess it makes sense that dependency imported only for their types should not implies that the associated styles should be part of the package. However this will be very complex to implement in the integrity test. So I propose to do it manually for this case (meaning we may be hit again by it at any point in the future).

Reply:
I can confirm that removing `@jupyterlab/application` from `@jupyterlab/lsp` resolves the issue.

cc @krassowski @hbcarlos what do you think?

Reply:
Maybe we do not need `@jupyterlab/application` here. The tracker could could just accept a shell-like object with an interface including `currentChanged` and `activeWidget`?

Reply:
Or could rely on trackers of individual document widgets passed during registration.

Reply:
> So I propose to do it manually for this case (meaning we may be hit again by it at any point in the future).

What do you mean by manually?  Adding `@jupyterlab/application` to the skip CSS here https://github.com/jupyterlab/jupyterlab/blob/97d320a850eaa5729d8a1d706799033ef19f31ac/buildutils/src/ensure-repo.ts#L192

so the integrity script doesn't try to include it?

Reply:
> so the integrity script doesn't try to include it?

Yes

---

> The tracker could just accept a shell-like object with an interface including currentChanged and activeWidget?

This would work too if it is defined locally within `@jupyterlab/lsp` as we cannot rely on `JupyterFrontEnd` as it is part of `@jupyterlab/application`.

Reply:
Sorry @krassowski, I didn't see your comments earlier.

> Maybe we do not need @jupyterlab/application here. The tracker could could just accept a shell-like object with an interface including currentChanged and activeWidget?

Sounds good. I implemented this solution in my last commit.